AppCenterDataStorage/AppCenterDataStorageTests/MSTokenExchangeTests.m

```diff
@@ -218,6 +217,48 @@ - (void)testValidCachedTokenExists {
   [utilityMock stopMocking];
 }
 
+- (void)testRetrieveCachedTokenFails {
+
+  // If
+  NSData *tokenData = [NSJSONSerialization dataWithJSONObject:[self getFailedTokenData] options:NSJSONWritingPrettyPrinted error:nil];
+  NSString *tokenString = [[NSString alloc] initWithData:tokenData encoding:NSUTF8StringEncoding];
+  OCMStub([self.keychainUtilMock stringForKey:mockTokenKeyName]).andReturn(tokenString);
+  id<MSHttpClientProtocol> httpMock = OCMProtocolMock(@protocol(MSHttpClientProtocol));
+
+  // Mock returning failed token.
+  NSObject *failedToken = [self getFailedTokenData];
+  NSMutableDictionary *tokenList = [@{kMSTokens : @[ failedToken ]} mutableCopy];
+  NSData *jsonTokenData = [NSJSONSerialization dataWithJSONObject:tokenList options:NSJSONWritingPrettyPrinted error:nil];
+  OCMStub([httpMock sendAsync:OCMOCK_ANY method:OCMOCK_ANY headers:OCMOCK_ANY data:OCMOCK_ANY completionHandler:OCMOCK_ANY])
+      .andDo(^(NSInvocation *invocation) {
+        __block MSHttpRequestCompletionHandler completionBlock;
+        [invocation getArgument:&completionBlock atIndex:6];
+        NSHTTPURLResponse *response = [NSHTTPURLResponse new];
+        id mockResponse = OCMPartialMock(response);
+        OCMStub([mockResponse statusCode]).andReturn(MSHTTPCodesNo200OK);
```

Comment:
Use `MSHttpTestUtil` to generate 200OK response instead of partial mocking response.